Please fix warnings in doc gen

```
Generating docs for file src//home/travis/build/cuhkshenzhen/CUHKSZLib/src/include/json/Json.h:277: warning: argument 'index' of command @param is not found in the argument list of JSONObject::at(const std::string &key)
/home/travis/build/cuhkshenzhen/CUHKSZLib/src/include/json/Json.h:282: warning: The following parameters of cuhksz::JSONObject::at(const std::string &key) are not documented:
  parameter 'key'
/home/travis/build/cuhkshenzhen/CUHKSZLib/src/include/json/Json.h:285: warning: argument 'index' of command @param is not found in the argument list of JSONObject::at(const std::string &key) const
/home/travis/build/cuhkshenzhen/CUHKSZLib/src/include/json/Json.h:289: warning: The following parameters of cuhksz::JSONObject::at(const std::string &key) const are not documented:
  parameter 'key'
/home/travis/build/cuhkshenzhen/CUHKSZLib/src/include/json/Json.h:200: warning: argument 'value' of command @param is not found in the argument list of JSONObject::operator=(T bool_value)
/home/travis/build/cuhkshenzhen/CUHKSZLib/src/include/json/Json.h:207: warning: The following parameters of cuhksz::JSONObject::operator=(T bool_value) are not documented:
  parameter 'bool_value'
```
@Yexiaoxing

--- a/codecov.yml
+++ b/codecov.yml
@@ -1,1 +1,4 @@
 comment: off
+ignore:
+  - tests/.*
+  - external/.*

--- a/src/include/json/Json.h
+++ b/src/include/json/Json.h
@@ -199,7 +199,7 @@ class JSONObject {
   /**
    * Load a bool into the JSONObject.
    * @tparam T Bool
-   * @param value Value of the bool variable
+   * @param bool_value Value of the bool variable
    * @return *this
    */
   template<typename T, typename std::enable_if<
@@ -275,15 +275,15 @@ class JSONObject {
   JSONObject &operator[](int index);
 
   /**
-   * Look up a value with specific index in a list. Will convert itself to an Object if not before.
-   * @param index index to look up
+   * Look up a value with specific key in a map object. Will convert itself to an Object if not before.
+   * @param key key to look up
    * @return Pointer to the value
    */
   JSONObject &at(const std::string &key);
 
   /**
-   * Look up a value with specific index in a list.
-   * @param index index to look up
+   * Look up a value with specific key in a map object.
+   * @param key key to look up
    * @return Pointer to the value
    */
   const JSONObject &at(const std::string &key) const;

--- a/src/json/Json.cpp
+++ b/src/json/Json.cpp
@@ -31,7 +31,7 @@ JSONObject parse_object(const std::string &str, size_t &offset) {
     skip_whitespaces(str, offset);
     if (str[offset] != ':') {
       error("ERROR: Object: Expected colon, found '" +
-            std::string(1, str[offset]) + "'");
+          std::string(1, str[offset]) + "'");
       break;
     }
     skip_whitespaces(str, ++offset);
@@ -47,7 +47,7 @@ JSONObject parse_object(const std::string &str, size_t &offset) {
       break;
     } else {
       error("ERROR: Object: Expected comma, found '" +
-            std::string(1, str[offset]) + "'");
+          std::string(1, str[offset]) + "'");
       break;
     }
   }
@@ -78,7 +78,7 @@ JSONObject parse_array(const std::string &str, size_t &offset) {
       break;
     } else {
       error("Array: Expected ',' or ']', found '" +
-            std::string(1, str[offset]) + "'");
+          std::string(1, str[offset]) + "'");
     }
   }
 
@@ -91,21 +91,29 @@ JSONObject parse_string(const std::string &str, size_t &offset) {
   for (char c = str[++offset]; c != '\"'; c = str[++offset]) {
     if (c == '\\') {
       switch (str[++offset]) {
-        case '\"':val += '\"';
+        case '\"':
+          val += '\"';
           break;
-        case '\\':val += '\\';
+        case '\\':
+          val += '\\';
           break;
-        case '/':val += '/';
+        case '/':
+          val += '/';
           break;
-        case 'b':val += '\b';
+        case 'b':
+          val += '\b';
           break;
-        case 'f':val += '\f';
+        case 'f':
+          val += '\f';
           break;
-        case 'n':val += '\n';
+        case 'n':
+          val += '\n';
           break;
-        case 'r':val += '\r';
+        case 'r':
+          val += '\r';
           break;
-        case 't':val += '\t';
+        case 't':
+          val += '\t';
           break;
         case 'u': {
           val += "\\u";
@@ -117,13 +125,14 @@ JSONObject parse_string(const std::string &str, size_t &offset) {
             } else {
               error(
                   "String: Expected hex character in unicode escape, found '" +
-                  std::string(1, c) + "'");
+                      std::string(1, c) + "'");
             }
           }
           offset += 4;
         }
           break;
-        default:val += '\\';
+        default:
+          val += '\\';
           break;
       }
     } else {
@@ -164,7 +173,7 @@ JSONObject parse_number(const std::string &str, size_t &offset) {
         exp_str += c;
       } else if (!isspace(c) && c != ',' && c != ']' && c != '}') {
         error("Number: Expected a number for exponent, found '" +
-              std::string(1, c) + "'");
+            std::string(1, c) + "'");
       } else {
         break;
       }
@@ -176,11 +185,11 @@ JSONObject parse_number(const std::string &str, size_t &offset) {
   --offset;
 
   if (isDouble)
-    Number = (int) cuhksz::stringCast<int>(val) * std::pow(10, exp);
+    Number = (int)cuhksz::stringCast<int>(val) * std::pow(10, exp);
   else if (!exp_str.empty())
-    Number = (int) cuhksz::stringCast<int>(val) * std::pow(10, exp);
+    Number = (int)cuhksz::stringCast<int>(val) * std::pow(10, exp);
   else
-    Number = (int) cuhksz::stringCast<int>(val);
+    Number = (int)cuhksz::stringCast<int>(val);
   return Number;
 }
 
@@ -192,7 +201,7 @@ JSONObject parse_bool(const std::string &str, size_t &offset) {
     Bool = false;
   } else {
     error("Bool: Expected 'true' or 'false', found '" + str.substr(offset, 5) +
-          "'");
+        "'");
   }
   offset += (Bool.toBool() ? 4 : 5);
   return Bool;
@@ -210,12 +219,17 @@ JSONObject parse_next(const std::string &str, size_t &offset) {
   skip_whitespaces(str, offset);
   value = str[offset];
   switch (value) {
-    case '[':return parse_array(str, offset);
-    case '{':return parse_object(str, offset);
-    case '\"':return parse_string(str, offset);
+    case '[':
+      return parse_array(str, offset);
+    case '{':
+      return parse_object(str, offset);
+    case '\"':
+      return parse_string(str, offset);
     case 't':
-    case 'f':return parse_bool(str, offset);
-    case 'n':return parse_null(str, offset);
+    case 'f':
+      return parse_bool(str, offset);
+    case 'n':
+      return parse_null(str, offset);
     default:
       if ((value <= '9' && value >= '0') || value == '-')
         return parse_number(str, offset);
@@ -245,19 +259,26 @@ void JSONObject::setType(Type type) {
   clearData();
 
   switch (type) {
-    case Type::Null:Data.Map = nullptr;
+    case Type::Null:
+      Data.Map = nullptr;
       break;
-    case Type::Object:Data.Map = new std::map<std::string, JSONObject>();
+    case Type::Object:
+      Data.Map = new std::map<std::string, JSONObject>();
       break;
-    case Type::Array:Data.List = new std::deque<JSONObject>();
+    case Type::Array:
+      Data.List = new std::deque<JSONObject>();
       break;
-    case Type::String:Data.String = new std::string();
+    case Type::String:
+      Data.String = new std::string();
       break;
-    case Type::Float:Data.Float = 0.0;
+    case Type::Float:
+      Data.Float = 0.0;
       break;
-    case Type::Integral:Data.Int = 0;
+    case Type::Integral:
+      Data.Int = 0;
       break;
-    case Type::Boolean:Data.Bool = false;
+    case Type::Boolean:
+      Data.Bool = false;
       break;
   }
   objType = type;
@@ -265,13 +286,17 @@ void JSONObject::setType(Type type) {
 
 void JSONObject::clearData() {
   switch (objType) {
-    case Type::Object:delete Data.Map;
+    case Type::Object:
+      delete Data.Map;
       break;
-    case Type::Array:delete Data.List;
+    case Type::Array:
+      delete Data.List;
       break;
-    case Type::String:delete Data.String;
+    case Type::String:
+      delete Data.String;
       break;
-    default: {}
+    default: {
+    }
   }
 }
 
@@ -285,9 +310,11 @@ JSONObject::JSONObject(const JSONObject &src) {
       Data.List = new std::deque<JSONObject>(src.Data.List->begin(),
                                              src.Data.List->end());
       break;
-    case Type::String:Data.String = new std::string(*src.Data.String);
+    case Type::String:
+      Data.String = new std::string(*src.Data.String);
       break;
-    default:Data = src.Data;
+    default:
+      Data = src.Data;
   }
   objType = src.objType;
 }
@@ -312,9 +339,11 @@ JSONObject &JSONObject::operator=(const JSONObject &src) {
       Data.List = new std::deque<JSONObject>(src.Data.List->begin(),
                                              src.Data.List->end());
       break;
-    case Type::String:Data.String = new std::string(*src.Data.String);
+    case Type::String:
+      Data.String = new std::string(*src.Data.String);
       break;
-    default:Data = src.Data;
+    default:
+      Data = src.Data;
   }
   objType = src.objType;
   return *this;
@@ -333,8 +362,8 @@ JSONObject &JSONObject::operator[](const std::string &key) {
 
 JSONObject &JSONObject::operator[](int index) {
   setType(Type::Array);
-  if (index >= (int) Data.List->size()) Data.List->resize((size_t) (index + 1));
-  return Data.List->operator[]((size_t) index);
+  if (index >= (int)Data.List->size()) Data.List->resize((size_t)(index + 1));
+  return Data.List->operator[]((size_t)index);
 }
 
 JSONObject &JSONObject::at(const std::string &key) { return operator[](key); }
@@ -350,7 +379,7 @@ const JSONObject &JSONObject::at(unsigned index) const {
 }
 
 int JSONObject::length() const {
-  return (int) (objType == Type::Array ? Data.List->size() : -1);
+  return (int)(objType == Type::Array ? Data.List->size() : -1);
 }
 
 bool JSONObject::hasKey(const std::string &key) const {
@@ -359,9 +388,9 @@ bool JSONObject::hasKey(const std::string &key) const {
 }
 
 int JSONObject::size() const {
-  return (int) (objType == Type::Object
-                ? Data.Map->size()
-                : objType == Type::Array ? Data.List->size() : -1);
+  return (int)(objType == Type::Object
+               ? Data.Map->size()
+               : objType == Type::Array ? Data.List->size() : -1);
 }
 
 std::string JSONObject::toString() const {
@@ -373,7 +402,7 @@ double JSONObject::toDouble() const {
 }
 
 float JSONObject::toFloat() const {
-  return (float) (objType == Type::Float ? Data.Float : 0.0);
+  return (float)(objType == Type::Float ? Data.Float : 0.0);
 }
 
 int JSONObject::toInt() const {
@@ -399,7 +428,8 @@ std::string JSONObject::dump(int depth, std::string tab) const {
   for (int i = 0; i < depth; ++i, pad += tab) {};
 
   switch (objType) {
-    case Type::Null:return "null";
+    case Type::Null:
+      return "null";
     case Type::Object: {
       std::string s = "{\n";
       bool skip = true;
@@ -422,10 +452,14 @@ std::string JSONObject::dump(int depth, std::string tab) const {
       s += "]";
       return s;
     }
-    case Type::String:return "\"" + jsonEscape(*Data.String) + "\"";
-    case Type::Float:return std::to_string(Data.Float);
-    case Type::Integral:return std::to_string(Data.Int);
-    case Type::Boolean:return Data.Bool ? "true" : "false";
+    case Type::String:
+      return "\"" + jsonEscape(*Data.String) + "\"";
+    case Type::Float:
+      return std::to_string(Data.Float);
+    case Type::Integral:
+      return std::to_string(Data.Int);
+    case Type::Boolean:
+      return Data.Bool ? "true" : "false";
   }
   return "";
 }

